docs/_build/html/_sources/methods.rst.txt

```diff
@@ -0,0 +1,444 @@
+.. set up matlab code highlighting
+.. highlight:: matlab
+
+.. set up referencing
+.. _methods:
+
+Methods
+=======
+
+This documentation will focus on using ``xolotl`` in ``MATLAB``, though it is
+possible to compile and run the ``C++`` code on its own.
+
+A ``xolotl`` object can be instantiated ::
+
+  x = xolotl;
+
+For consistency, this documentation will assume a ``xolotl`` object named ``x``.
+
+.. hint::
+
+  You can list all the methods of ``xolotl`` object ``x`` by ::
+
+    x.methods
+
+.. _children:
+
+Children
+^^^^^^^^
+
+Returns the children of any ``cpplab`` object as a cell of character vectors.
+Children are also ``cpplab`` objects. For example, if ``HH`` is the only compartment in ``x`` then ::
+
+  x.Children
+
+returns ::
+
+  {'HH'}
+
+This method is inherited from the superclass ``cpplab``.
+
+.. _add:
+
+add
+^^^
+
+Adds a compartment, conductance, or synapse to the ``xolotl`` object. To add
+a compartment::
+
+  x.add('compartmentName','compartment','PropertyName','PropertyValue', ...)
+
+Valid compartment properties are
+
+======================= ================
+Name                    ``PropertyName``
+Volume                  ``vol``
+Surface area            ``A``
+Extra. calcium steady.  ``Ca_out``
+Buffer shell thickness  ``shell_thickness``
+Intra. calcium steady.  ``Ca_in``
+Ca++ curr. to conc.     ``phi``
+Average Ca++ conc.      ``Ca_average``
+Radius of compartment   ``radius``
+Target Ca++ conc.       ``Ca_target``
+Index in `xolotl` tree  ``tree_idx``
+Index of neuron         ``neuron_idx``
+Length of compartment   ``len``
+Ca++ buff. time const.  ``tau_Ca``
+Membrane capacitance    ``Cm``
+Intra. calcium conc.    ``Ca``
+Membrane potential      ``V``
+======================= ================
+
+To add a compartment named ``HH`` with a membrane capacitance ``Cm = 10`` ::
+
+  x.add('HH', 'compartment', 'Cm', 10);
+
+To access properties of that compartment, use the syntax ``x.HH``. To add a conductance to that compartment ::
+
+  x.HH.add('ConductancePath', 'PropertyName', 'PropertyValue', ...);
+
+``ConductancePath`` is the path to the ``.hpp`` file inside ``xolotl/c++/conductances/``. For example, the conductance path to add the slow calcium conductance from Prinz *et al.* 2003 is ``prinz/CaS`` since the conductance is specified in ``xolotl/c++/conductances/prinz/CaS.hpp``.
```

Comment:
it doesn't have to be the exact path. any fragment of the path will be matched.

Reply:
done